pkg/gardenlet/controller/federatedseed/network-policy/endpoints_controller.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package network_policy
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/gardenlet/controller/federatedseed/network-policy/helper"
+
+	"k8s.io/client-go/tools/cache"
+)
+
+func (c *Controller) endpointAdd(_ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) endpointUpdate(_, _ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) endpointDelete(_ interface{}) {
+	c.enqueueNamespaces()
+}
+
+func (c *Controller) enqueueNamespaces() {
+	namespaces, err := c.namespaceLister.List(c.shootNamespaceSelector)
+	if err != nil {
+		c.log.Errorf("Failed to enqueue namespaces to update NetworkPolicy %q - unable to list Shoot namespaces: %v", helper.AllowToSeedAPIServer, err)
+		return
+	}
+
+	gardenNamespace, err := c.namespaceLister.Get(v1beta1constants.GardenNamespace)
+	if client.IgnoreNotFound(err) != nil {
+		c.log.Errorf("Failed to enqueue namespaces to update NetworkPolicy %q: %v", helper.AllowToSeedAPIServer, err)
+		return
+	}
+	if gardenNamespace != nil {
+		namespaces = append(namespaces, gardenNamespace)
+	}
```

Comment:
Can't you always add the `garden` namespace? Usually, the reconciler properly reacts on `NotFound` errors when trying to read the queued object from the cache. (Anyways, there will always be a `garden` namespace in the seed.)